test/actions/helpers.js

```diff
@@ -1,44 +1,215 @@
+  };
+
+  _config.apiRoot = 'http://localhost';
+});
+
+test('formatError() should handle error responses properly', (t) => {
+  const requestErrorMessage = 'Request failed';
+  const response = {
+    statusMessage: requestErrorMessage
+  };
+
+  let formattedError = formatError();
+  t.is(formattedError, '');
+
+  formattedError = formatError(null);
+  t.is(formattedError, '');
+
+  formattedError = formatError(response);
+  t.is(formattedError, requestErrorMessage);
+
+  const bodyErrorName = 'InternalServerError';
+  const bodyErrorMessage = 'Internal server error';
+
+  formattedError = formatError(null, {
+    name: bodyErrorName
+  });
+  t.is(formattedError, bodyErrorName);
+
+  formattedError = formatError(null, {
+    message: bodyErrorMessage
+  });
+  t.is(formattedError, bodyErrorMessage);
+
+  formattedError = formatError(response, {
+    name: bodyErrorName
+  });
+  t.is(formattedError, bodyErrorName);
+
+  formattedError = formatError(response, {
+    message: bodyErrorMessage
+  });
+  t.is(formattedError, `${requestErrorMessage}: ${bodyErrorMessage}`);
+
+  formattedError = formatError(response, {
+    name: bodyErrorName,
+    message: bodyErrorMessage
+  });
+  t.is(formattedError, `${bodyErrorName}: ${bodyErrorMessage}`);
+});
+
+test('addRequestAuthorization() should return correct headers', (t) => {
+  const config = {};
+  addRequestAuthorization(config, getStateStub());
+  t.deepEqual(config.headers, {
+    Authorization: 'Bearer fake-token'
+  });
+});
+
+test('configureRequest() should throw error if no URL or path is provided', (t) => {
+  const requestParams = {};
+
+  t.throws(() => configureRequest(requestParams), 'Must include a URL or path with request');
+});
+
+test('configureRequest() should throw error if path is not a string', (t) => {
+  const requestParams = { path: {} };
+
+  t.throws(() => configureRequest(requestParams), 'Path must be a string');
+});
+
+test('configureRequest() should add default parameters', (t) => {
+  const requestParams = {
+    url: 'http://localhost/test'
+  };
+  const expectedConfig = {
+    ...t.context.defaultConfig,
+    url: 'http://localhost/test',
+    headers: t.context.defaultHeaders
+  };
+  const requestConfig = configureRequest(requestParams);
+  t.deepEqual(requestConfig, expectedConfig);
+});
+
+test('configureRequest() should convert path to URL', (t) => {
+  const requestParams = {
+    path: 'test'
+  };
+  const expectedConfig = {
+    ...t.context.defaultConfig,
+    path: 'test',
+    url: 'http://localhost/test',
+    headers: t.context.defaultHeaders
+  };
+  const requestConfig = configureRequest(requestParams);
+  t.deepEqual(requestConfig, expectedConfig);
+});
+
+test('configureRequest() should maintain the request body', (t) => {
+  const requestBody = {
+    test: 'test'
+  };
+  const requestParams = {
+    url: 'http://localhost/test',
+    body: requestBody
+  };
+  const expectedConfig = {
+    ...t.context.defaultConfig,
```

Comment:
Is this value transformed into an array object at some point? I didn't think the spread operator works with dictionaries...

Reply:
Yup, the spread operator does work on object literals: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals